MathUtils.hpp

```diff
@@ -36,7 +37,9 @@ namespace quickhull {
 			T px = y * rhsz - z * rhsy ;
 			T py = z * rhsx - x * rhsz ;
 			T pz = x * rhsy - y * rhsx ;
-			return Vector3<T>(px,py,pz);
+      // errorNormal = Vector3<T>(std::abs(y*rhsz)+std::abs(z*rhsy),std::abs(z*rhsx)+std::abs(x*rhsz),std::abs(x*rhsy)+std::abs(y*rhsx));
+			// return Vector3<T>(px,py,pz);
+      return Vector3<T>(px,py,pz).getNormalized();
```

Comment:
I think this is your proposed fix?

Reply:
Yes, so basically normalizing the output of getTriangleNormal function, since we were comparing the distance (unnormalized) directly to epsilon in 
https://github.com/akuukka/quickhull/blob/4ef66c68950cb4db11d3b75bfe4034d807485ad0/QuickHull.cpp#L462 [maxD was initially set to m_epsilon] 
and based on my understanding of the code that wasn't intended.

Also, the above line and 
https://github.com/akuukka/quickhull/blob/4ef66c68950cb4db11d3b75bfe4034d807485ad0/QuickHull.hpp#L196-L206
Both compare distances to find the maximum distance, in such a situation when the distance is not the Euclidean distance the comparison could select an "incorrect point" (Now I know that we had discussed that choosing the absolute maximum point doesn't make a difference in floating point arithmetic, but it's possible that due to this non-Euclidean distance we pick a point which is very close, which could lead to the errors [this is just speculation on my end]  leading to those errors)

PS: I'm sorry if I wasn't able to explain this properly in the original comment. I hope it's clearer now

Reply:
Good find! Since normalization is somewhat expensive, you might want to check all the places this function is called. If they all need to be normalized, then this is perfect, but if not, it would be ideal to only add normalization where needed. Have you noticed any difference in speed from this change?